cmd/main.go

```diff
@@ -302,18 +303,28 @@ func main() {
 		crdUtils = controllers.NewDummyCRDUtils()
 	}
 
-	bm := bminventory.NewBareMetalInventory(db, log.WithField("pkg", "Inventory"), hostApi, clusterApi, Options.BMConfig,
-		generator, eventsHandler, objectHandler, metricsManager, operatorsManager, authHandler, ocpClient, ocmClient,
-		lead, pullSecretValidator, versionHandler, isoEditorFactory, crdUtils, staticNetworkConfig)
+	gc := common.NewGarbageCollectors(db, log.WithField("pkg", "Inventory"), hostApi, clusterApi, objectHandler, lead)
+
+	deregisterWorker := thread.New(
+		log.WithField("inventory", "Deregister Worker"),
+		"Deregister Worker",
+		Options.DeregisterWorkerInterval,
+		gc.PermanentlyDeleteUnregisteredClustersAndHosts)
```

Comment:
```suggestion
		gc.InactiveClusterDeregister)
```
Why is it the same function in two different threads? 
Shouldn't one just delete the clusters and the other clear old deleted clusters?